cvat-sdk/cvat_sdk/core/client.py

```diff
@@ -253,25 +253,27 @@ def check_server_version(self, fail_if_unsupported: Optional[bool] = None) -> No
                 raise IncompatibleVersionException(msg)
             return
 
-        sdk_version = pv.Version(VERSION)
-
-        # We only check base version match. Micro releases and fixes do not affect
-        # API compatibility in general.
-        if all(
-            server_version.base_version != sv.base_version for sv in self.SUPPORTED_SERVER_VERSIONS
+        if not any(
+            self._is_version_compatible(server_version, supported_version)
+            for supported_version in self.SUPPORTED_SERVER_VERSIONS
         ):
             msg = (
                 "Server version '%s' is not compatible with SDK version '%s'. "
                 "Some SDK functions may not work properly with this server. "
                 "You can continue using this SDK, or you can "
                 "try to update with 'pip install cvat-sdk'."
-            ) % (server_version, sdk_version)
+            ) % (server_version, pv.Version(VERSION))
             self.logger.warning(msg)
             if fail_if_unsupported:
                 raise IncompatibleVersionException(msg)
 
+    def _is_version_compatible(self, current: pv.Version, target: pv.Version) -> bool:
+        # Check for (major, minor) compatibility.
+        # Micro releases and fixes do not affect API compatibility in general.
+        upper_bound = pv.Version(f"{target.epoch}{target.major}.{target.minor + 1}")
+        return target <= current < upper_bound
```

Comment:
I think this is slightly wrong, because it will calculate that 2.5.0a1 is compatible with 2.4.0. I would suggest something like this instead:

```
return current in packaging.specifiers.SpecifierSet(f"~= {target.epoch}!{target.major}.{target.minor}.{target.micro}")
```

Reply:
Yes, it's a nice catch.

Reply:
Since the version on the right side comes from the SDK, I wouldn't expect it to be a prerelease. And what's wrong with the 3rd case?

Reply:
I think, given what we do now (add the next release version to support the public instance, and once we publish it, we update SDK), it makes sense to try to limit the next version to its pre-release variant, so the old SDK will stop working automatically, once the new server release is deployed.

E.g.:
public server: 2.5.devX
sdk: v2.4 supporting >=2.4, <2.5

public server: 2.5
sdk v2.4 stops working